sklearn/metrics/_plot/regression.py

```diff
@@ -0,0 +1,409 @@
+import numbers
+
+import numpy as np
+
+from ...utils import check_matplotlib_support
+from ...utils import check_random_state
+from ...utils import _safe_indexing
+
+
+class PredictionErrorDisplay:
+    """Prediction error visualization.
+
+    It is recommended to use
+    :func:`~sklearn.metrics.PredictionErrorDisplay.from_estimator` or
+    :func:`~sklearn.metrics.PredictionErrorDisplay.from_predictions` to
+    create a visualizer. All parameters are stored as attributes.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    .. versionadded:: 1.2
+
+    Parameters
+    ----------
+    y_true : ndarray of shape (n_samples,)
+        True values.
+
+    y_pred : ndarray of shape (n_samples,)
+        Prediction values.
+
+    Attributes
+    ----------
+    line_ : matplotlib Artist
+        Optimal line representing `y_true == y_pred`. Therefore, it is a
+        diagonal line for `kind="predictions"` and a horizontal line for
+        `kind="residuals"`.
+
+    errors_lines_ : matplotlib Artist or None
+        Residual lines. If `with_errors=False`, then it is set to `None`.
+
+    scatter_ : matplotlib Artist
+        Scatter data points.
+
+    ax_ : matplotlib Axes
+        Axes with the different matplotlib axis.
+
+    figure_ : matplotlib Figure
+        Figure containing the scatter and lines.
+
+    See Also
+    --------
+    PredictionErrorDisplay.from_estimator : Prediction error visualization
+        given an estimator and some data.
+    PredictionErrorDisplay.from_predictions : Prediction error visualization
+        given the true and predicted targets.
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt
+    >>> from sklearn.datasets import load_diabetes
+    >>> from sklearn.linear_model import Ridge
+    >>> from sklearn.metrics import PredictionErrorDisplay
+    >>> X, y = load_diabetes(return_X_y=True)
+    >>> ridge = Ridge().fit(X, y)
+    >>> y_pred = ridge.predict(X)
+    >>> display = PredictionErrorDisplay(y_true=y, y_pred=y_pred)
+    >>> display.plot()
+    <...>
+    >>> plt.show()
+    """
+
+    def __init__(self, *, y_true, y_pred):
+        self.y_true = y_true
+        self.y_pred = y_pred
+
+    def plot(
+        self,
+        ax=None,
+        *,
+        kind="predictions",
+        x_axis="actual_targets",
+        scatter_kwargs=None,
+        line_kwargs=None,
+    ):
+        """Plot visualization.
+
+        Extra keyword arguments will be passed to matplotlib's ``plot``.
+
+        Parameters
+        ----------
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        kind : {"predictions", "residuals"}, default="predictions"
+            The type of plot to draw:
+
+            - "predictions" draws the predicted values vs. the true values.
```

Comment:
This should be true values (y axis) vs predict values (x axis). But we also need to check that this is actually what is done in the code.

Reply:
This does not seem to be the case when looking at the figures from the example included in the user guide.

Reply:
I searched widely and I could not find a clear consensus on plotting "predicted vs. actual" or "actual vs. predicted". Up to now, we always used "predicted (y-axis) vs. actual (x-axis)" in scikit-learn.

Since we actually have the same data, I would propose removing the parameter `x_axis` and going with "actual vs. predicted" to have always "predicted" in x-axis, as originally proposed by @lorentzenchr

Reply:
See Glen_b’s comment in https://stats.stackexchange.com/questions/104622/what-does-an-actual-vs-fitted-graph-tell-us
> It would be more common to follow the convention of plotting the values that are fixed (conditional on predictors) on the x-axis and the values that are random on the y-axis.

I had a hard time finding this plot in books at all😉